platform-sdk/docs/proposals/TSS-Ledger-Id/TSS-Ledger-Id.md

```diff
@@ -0,0 +1,826 @@
+| 6 | Retrospectively sign the initial blocks with TSS once it becomes available.                                   | <span style="color:green">YES</span> | <span style="color:green">YES</span>                 | <span style="color:green">YES</span> | <span style="color:green">YES</span> | <span style="color:green">YES</span> | <span style="color:green">YES</span> | <span style="color:green">YES</span> | <span style="color:red">HIGH</span>       | TSS Genesis Roster signs older blocks.  This is compatible with all other options. |
+| 7 | Pre-Genesis: Separate app with genesis roster creates key material in state before network officially starts. | <span style="color:green">YES</span> | <span style="color:green">YES</span>                 | <span style="color:green">YES</span> | <span style="color:green">YES</span> | <span style="color:red">NO</span>    | <span style="color:red">NO</span>    | <span style="color:red">NO</span>    | <span style="color:red">HIGH</span>       | Applies to new networks only.  Use Option 1 or 2 for existing networks.            |
+| 8 | Instead of separate app, detect genesis, key the state, network restart from round 0 with keyed state.        | <span style="color:green">YES</span> | <span style="color:green">YES</span>                 | <span style="color:green">YES</span> | <span style="color:green">YES</span> | <span style="color:red">NO</span>    | <span style="color:red">NO</span>    | <span style="color:green">YES</span> | <span style="color:red">HIGH</span>       | Applies to new networks only.  Use Option 1 or 2 for existing networks.            | 
+
+Option 2 was chosen for existing networks and Option 8 was selected for new networks. This decision was made to
+ensure that the first block in the block stream is signed by the ledger and that new networks always have block 0
+signed.
+
+---
+
+## Changes
+
+The changes are presented in the following order:
+
+1. Public API
+2. Core Behaviors
+3. Component Architecture
+4. Configuration
+5. Metrics
+6. Performance
+
+### Public API
+
+The public API changes consist of the following:
+
+* The Ledger Signature API
+* System Transactions
+* State Data Structures
+
+#### Ledger Signature API
+
+The Ledger Signature API is the interface between the application and the platform by which the application submits
+a message to be signed by the ledger, the nodes sign the message with their shares, and the platform aggregates the
+signatures into a ledger signature. The ledger signature is then asynchronously returned to the application.
+
+To request a ledger signature, the application invokes a new method on the `State`:
+
+* `State::sign(byte[] message) throws NotReadyException`
+
+To receive a ledger signature, the application registers a callback with the platform builder:
+
+* `PlatformBuilder::withLedgerSignatureCallback(Consumer<PairingSignature> ledgerSignatureConsumer)`
```

Comment:
Is the reason for communicating the signature outside of the state where the application can read the value on the handle thread so that if consensus is paused or stops, or if we re-implement quiescence, the signature is still provided? If so, i think that explanation should be provided.